api/src/oscc.c

```diff
@@ -1092,3 +1092,84 @@ oscc_result_t get_device_name( char * string, char * const name )
 
     return result;
 }
+
+oscc_result_t get_wheel_speed_right_rear(
+    struct can_frame const * const frame,
+    double * wheel_speed_right_rear)
+{
+    if((frame == NULL) || (wheel_speed_right_rear == NULL)) { return OSCC_ERROR; }
+    if(frame->can_id != KIA_SOUL_OBD_WHEEL_SPEED_CAN_ID) { return OSCC_ERROR; }
+    uint16_t raw = ((frame->data[7] & 0x0F) << 8) | frame->data[6];
+    // 10^-1 precision, raw / 32.0
+    *wheel_speed_right_rear = (double)((int)((double)raw / 3.2) / 10.0);
+    return OSCC_OK;
+}
+
+
+oscc_result_t get_wheel_speed_left_rear(
+    struct can_frame const * const frame,
+    double * wheel_speed_left_rear)
+{
+    if((frame == NULL) || (wheel_speed_left_rear == NULL)) { return OSCC_ERROR; }
+    if(frame->can_id != KIA_SOUL_OBD_WHEEL_SPEED_CAN_ID) { return OSCC_ERROR; }
+    uint16_t raw = ((frame->data[5] & 0x0F) << 8) | frame->data[4];
+    // 10^-1 precision, raw / 32.0
+    *wheel_speed_left_rear = (double)((int)((double)raw / 3.2) / 10.0);
+    return OSCC_OK;
+}
+
+
+oscc_result_t get_wheel_speed_right_front(
+    struct can_frame const * const frame,
+    double * wheel_speed_right_front)
+{
+    if((frame == NULL) || (wheel_speed_right_front == NULL)) { return OSCC_ERROR; }
+    if(frame->can_id != KIA_SOUL_OBD_WHEEL_SPEED_CAN_ID) { return OSCC_ERROR; }
+    uint16_t raw = ((frame->data[3] & 0x0F) << 8) | frame->data[2];
+    // 10^-1 precision, raw / 32.0
+    *wheel_speed_right_front = (double)((int)((double)raw / 3.2) / 10.0);
+    return OSCC_OK;
+}
+
+
+oscc_result_t get_wheel_speed_left_front(
+    struct can_frame const * const frame,
+    double * wheel_speed_left_front)
+{
+    if((frame == NULL) || (wheel_speed_left_front == NULL)) { return OSCC_ERROR; }
+    if(frame->can_id != KIA_SOUL_OBD_WHEEL_SPEED_CAN_ID) { return OSCC_ERROR; }
+    uint16_t raw = ((frame->data[1] & 0x0F) << 8) | frame->data[0];
+    // 10^-1 precision, raw / 32.0
+    *wheel_speed_left_front = (double)((int)((double)raw / 3.2) / 10.0);
+    return OSCC_OK;
+}
+
+
+oscc_result_t get_steering_wheel_angle(
+    struct can_frame const * const frame,
+    double * steering_wheel_angle)
+{
+    if((frame == NULL) || (steering_wheel_angle == NULL)) { return OSCC_ERROR; }
+    if(frame->can_id != KIA_SOUL_OBD_STEERING_WHEEL_ANGLE_CAN_ID) { return OSCC_ERROR; }
+    int16_t raw = (frame->data[1] << 8) | frame->data[0];
+    *steering_wheel_angle = -((double)raw / 10.0);
```

Comment:
I believe that we have the [`KIA_SOUL_OBD_STEERING_ANGLE_SCALAR`](https://github.com/PolySync/oscc/blob/master/api/include/vehicles/kia_niro.h#L62) that we used for this conversion in the past, though it did not take into consideration the sign reversal. Maybe we could deprecate that constant (for each vehicle) as well?